src/pages/login.astro

```diff
@@ -67,29 +67,35 @@ if (Astro.cookies.has("__session")) {
 
     // Inside login.astro script section, modify the event listener:
     loginForm.addEventListener("submit", async (event) => {
-    event.preventDefault();
+  event.preventDefault();
 
-    const formData = new FormData(loginForm);
-    const email = formData.get("email")?.toString();
-    const password = formData.get("password")?.toString();
+  const formData = new FormData(loginForm);
+  const email = formData.get("email")?.toString();
+  const password = formData.get("password")?.toString();
 
-    if (!email || !password) {
-      errorMessageEl.textContent = "Please fill in all fields";
+  if (!email || !password) {
+    errorMessageEl.textContent = "Please fill in all fields";
+    return;
+  }
+
+  try {
+    const userCredential = await signInWithEmailAndPassword(auth, email, password);
+    
+    // Check if email is verified
+    if (!userCredential.user.emailVerified) {
+      errorMessageEl.textContent = "Please verify your email before logging in.";
       return;
     }
 
-    try {
-      const userCredential = await signInWithEmailAndPassword(auth, email, password);
-      const idToken = await userCredential.user.getIdToken();
-
-      await actions.login({ idToken });
+    const idToken = await userCredential.user.getIdToken();
+    await actions.login({ idToken });
```

Comment:
The server side action also needs to check that the user is email verified (i.e. currently a malicious client could remove the client-side log-in check and call the login action with an ID Token for an unverified account).